### PR TITLE
Check goal status before cancelling. Fixes #71.

### LINKED ISSUE #71: SimpleActionClient error when requesting move in paused state
### Commit
0e3b55f5272c7609b3822a937915dafcb5ac7070

### Steps to reproduce
See `acceptance_test_move_ctrl.py`

### Observed behavior
An error is displayed in the test `_test_pause_between_moves`

### Logfiles

> [INFO] [WallTime: 1548918997.828340 /home/imartini/ros/src/pilz_industrial_motion/pilz_robot_programming/src/pilz_robot_programming/robot.py 245]: Pause called.
[INFO] [WallTime: 1548918997.828578 /home/imartini/ros/src/pilz_industrial_motion/pilz_robot_programming/src/pilz_robot_programming/move_control_request.py 98]: Switching state from NO_REQUEST to PAUSE_REQUESTED
[INFO] [WallTime: 1548918997.829016 /home/imartini/ros/src/pilz_industrial_motion/pilz_robot_programming/src/pilz_robot_programming/robot.py 208]: Move: Ptp
[INFO] [WallTime: 1548918997.829279 /home/imartini/ros/src/pilz_industrial_motion/pilz_robot_programming/src/pilz_robot_programming/robot.py 307]: start wait for resume
[ERROR] [WallTime: 1548918997.979876 /opt/ros/kinetic/lib/python2.7/dist-packages/actionlib/simple_action_client.py 237]: Received comm state PREEMPTING when in simple state DONE with SimpleActionClient in NS /sequence_move_group
[INFO] [WallTime: 1548919002.829566 /home/imartini/ros/src/pilz_industrial_motion/pilz_robot_programming/src/pilz_robot_programming/robot.py 259]: Resume called.

### PR DESCRIPTION
**Pull request by martiniil**
There still is a time window, where the error can happen. This is the best we can do to prevent it.

### CODE REVIEW

File: pilz_robot_programming/src/pilz_robot_programming/robot.py [hunk 1]
```diff
@@ -237,7 +240,11 @@ def stop(self):
         """
         rospy.loginfo("Stop called.")
         self._move_ctrl_sm.switch(MoveControlAction.STOP)
-        self._cancel_on_all_clients()
+
+        with self._move_ctrl_sm: # wait, if _execute is just starting a send_goal()
+            actionclient_state = self._sequence_client.get_state()
+            if any(state == actionclient_state for state in _VALID_GOAL_STATUS_FOR_CANCEL):
```
Comment (agutenkunst) [inline pilz_robot_programming/src/pilz_robot_programming/robot.py]: Wouldn't `if actionclient_state in _VALID_GOAL_STATUS_FOR_CANCEL` work?

**Review by agutenkunst — CHANGES_REQUESTED** [review]

File: pilz_robot_programming/src/pilz_robot_programming/robot.py (hunk 1 above)
Comment (martiniil) [reply pilz_robot_programming/src/pilz_robot_programming/robot.py]: It works.

**Review by agutenkunst — APPROVED** [review]

**Comment by jschleicher** [comment]
@martiniil Is there any chance of an upstream fix / PR?

**Comment by martiniil** [comment]
@jschleicher Maybe, the problem is not that I would not like to do it ...

**Pull request merged**